operator/scheduler/model.go

```diff
@@ -161,10 +161,12 @@ func (s *SchedulerClient) SubscribeModelEvents(ctx context.Context, grpcClient s
 		// Handle terminated event to remove finalizer
 		if canRemoveFinalizer(latestVersionStatus.State.State) {
 			retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-				latestModel := &v1alpha1.Model{}
+				ctxWithTimeout, cancel := context.WithTimeout(ctx, constants.K8sAPICallTimeout)
```

Comment:
One thing I've realised here is that, despite the name `K8sAPICallTimeout` covers not just one k8s API call but all calls done in one function. So in the example here, both the `s.Get(...)` and `s.Update(...)` need to complete within a total time < than the timeout.

Reply:
This is the intention to simplify things. I can come up with a better naming for the constant to make it clearer.